nemo_aligner/data/nlp/builders.py

```diff
@@ -339,16 +340,12 @@ def build_dataloader(
         "pad_samples_to_global_batch_size": pad_samples_to_global_batch_size,
     }
 
-    # Megatron sampler
-    if hasattr(cfg.model.data, "dataloader_type") and cfg.model.data.dataloader_type == "single":
```

Comment:
actually cyclic and using random samplers has an overlap. In NeMo setting dataloader type to cyclic means using the random dataloader [here](https://github.com/NVIDIA/NeMo/blob/e68f981c393441165548e45c65c49ed5283fc0d5/nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py#L1648) so I found this option to be redundant in aligner. But it is indeed true it doesn't matter what the user set(cyclic or not cyclic) since the random sampler behavior will overwrite it anyway

Reply:
I feel like(maybe in a future PR): we should just functionalize the `build_dataloader` function so don't have this confusion. currently the cfg is really just used for the seed and num workers

Reply:
created an issue: https://github.com/NVIDIA/NeMo-Aligner/issues/273

Reply:
> it doesn't matter what the user set(cyclic or not cyclic) since the random sampler behavior will overwrite it anyway

Should we just remove this option from our configs then?

Reply:
done! 02eccd5f54fa3ec74bb95234d557500eeb6ab167